### PR TITLE
Define `__str__` for `Device` classes

### PR DESCRIPTION
**Pull request by nishnik**
This PR addresses #6772 

I am not sure if this would need a unit test. Please tell if any modification is required.
@niboshi

### CODE REVIEW

File: chainer/backends/_chainerx.py [hunk 1]
```diff
@@ -65,6 +65,10 @@ def __repr__(self):
         return '<{} {}>'.format(
             self.__class__.__name__, self.device.name)
 
+    def __str__(self):
+        return '{}: {}'.format(
+            self.__class__.__name__, self.device.name)
```
Comment (niboshi) [inline chainer/backends/_chainerx.py]: Sorry it might not be clear, but the class name part is not needed: just return `self.device.name` in this case.

**Comment by niboshi** [comment]
Thank you for PR!

Yes, please add unit tests.

In particular, please add `assert` statements in  `TestDevice.test_repr_str_...` in `tests/chainer_tests/test_backend.py` (replacing `str` with `repr` in existing asserts).

**Comment by nishnik** [comment]
I have made the aforementioned changes, and have added the tests. Please have a look @niboshi

File: chainer/backends/_chainerx.py [hunk 2]
```diff
@@ -65,6 +65,9 @@ def __repr__(self):
         return '<{} {}>'.format(
             self.__class__.__name__, self.device.name)
 
+    def __str__(self):
+        return '{}'.format(self.device.name)
```
Comment (niboshi) [inline chainer/backends/_chainerx.py]: Simpler:
```suggestion
        return self.device.name
```

File: chainer/backends/_cpu.py [hunk 3]
```diff
@@ -30,6 +30,9 @@ def __eq__(self, other):
     def __repr__(self):
         return '<{} (numpy)>'.format(self.__class__.__name__)
 
+    def __str__(self):
+        return '{}: @numpy'.format(self.__class__.__name__)
```
Comment (niboshi) [inline chainer/backends/_cpu.py]: Please do not include class name.
```suggestion
        return '@numpy'
```

File: chainer/backends/cuda.py [hunk 4]
```diff
@@ -231,6 +231,10 @@ def __repr__(self):
         return '<{} (cupy):{}>'.format(
             self.__class__.__name__, self.device.id)
 
+    def __str__(self):
+        return '{}: @cupy:{}'.format(
+            self.__class__.__name__, self.device.id)
```
Comment (niboshi) [inline chainer/backends/cuda.py]: Ditto. Please do not include clas name.

File: chainer/backends/intel64.py [hunk 5]
```diff
@@ -49,6 +49,9 @@ def __eq__(self, other):
     def __repr__(self):
         return '<{}>'.format(self.__class__.__name__)
 
+    def __str__(self):
+        return '{}: @intel64'.format(self.__class__.__name__)
```
Comment (niboshi) [inline chainer/backends/intel64.py]: Ditto.
```suggestion
        return '@intel64'
```

**Comment by nishnik** [comment]
Please review @niboshi 
Thanks!

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM
Thank you so much!

**Pull request merged**